builtin/providers/aws/resource_aws_ecs_service.go

```diff
@@ -241,21 +241,18 @@ func resourceAwsEcsServiceUpdate(d *schema.ResourceData, meta interface{}) error
 func resourceAwsEcsServiceDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ecsconn
 
+	serviceName := d.Id()
+	clusterName := d.Get("cluster").(string)
+
 	// Check if it's not already gone
 	resp, err := conn.DescribeServices(&ecs.DescribeServicesInput{
-		Services: []*string{aws.String(d.Id())},
-		Cluster:  aws.String(d.Get("cluster").(string)),
+		Services: []*string{aws.String(serviceName)},
```

Comment:
ARNs are AFAIK the most specific ids and also may help us while debugging some issues related to region or account ID - imagine you would use wrong AWS credentials and then you'd be wondering why the ECS service is/isn't there or why it has different settings.

For those reasons I'd be personally inclined to keep ARNs where possible, especially in logs.